src/ai/missions/MiningMission.ts

```diff
@@ -333,6 +333,13 @@ export class MiningMission extends Mission {
 
     private placeContainer() {
 
+        if (this.room.controller.reservation &&
+            /* reserved and not mine */
+            this.room.controller.reservation.username != Game.structures[_.first(Object.keys(Game.structures))].owner.username) {
```

Comment:
I expect bonzAI has a better way to get the player's username than this, but it didn't jump out at me.

Reply:
The constant USERNAME evaluates to the player's username, which is assigned at global refresh time (basically using the same method here)